src/tool_storage.rs

```diff
@@ -154,7 +159,7 @@ impl ToolStorage {
         log::info!("Installing tool: {}", spec);
 
         log::debug!("Fetching GitHub releases...");
-        let github = self.github.get_or_init(GitHubSource::new);
+        let github = self.github.get_or_init(|| GitHubSource::new(self.auth));
```

Comment:
> I've attempted to fix this, but will need some help with borrowing as I'm still new to rust

You will need to clone here to not move ownership of the personal access token out of `AuthManifest`, or you could make `GitHubSource` take a `&str` instead of `String` but that might be unnecessary and painful to deal with

```suggestion
        let github = self.github.get_or_init(|| {
            GitHubSource::new(match &self.auth {
                Some(auth) => auth.github.clone(),
                None => None,
            })
        });
```